ClinGen/clincoded
# Connects to #974. Connects to #836. Unify status behavior. Update history tracking and display

#### Connects to #836:

* Note: Was connected to #839 as well
* Note: Both tickets were made somewhat outdated by changes to the interpretation list format in other tickets/PRs
* Puts ORPHA### in parentheses instead of after a comma 

#### Connects to #974:

* Changed calculated status for interpretations: 'provisional' if there is provisional classification; 'in progress' in all other cases (0 or more interpretations saved)
* Action of adding interpretation now shows up on dashboard history
* Variant w/ no evaluations now show up on variant list under 'My Variants' list on dashboard

#### Others:

* Updated interpretation history item to better track MOI
* Updated the way interpretation history items are created to save snapshot of interpretation title at moment of history creation
* Updated the way interpretation items are displayed on Dashboard (My Interpretation)
* Updated the way interpretation-related items are displayed on Dashboard (Recent History)


*Testing:*

1. Enter VCI. Add Interpretation to Variant
  * Confirm that Variant now shows up on Dashboard's Recent History and My Interpretations list
  * Confirm that Variant status is 'in progress'
2. Add Disease to Variant
  * Confirm that action shows up on Dashboard's Recent History w/ correct Variant title
  * Confirm that the Variant's Disease field updated properly under Dashboard's My Interpretation
3. Update evaluation, and confirm that the action shows up on Dashboard's Recent History w/ correct Variant title
4. Add Mode of Inheritance to Variant
  * Confirm that action shows up on Dashboard's Recent History w/ correct Variant title
  * Confirm that the Variant's Mode of Inheritance field updated properly under Dashboard's My Interpretation
5. Update evaluation, and confirm that the action shows up on Dashboard's Recent History w/ correct Variant title
6. Add Provisional Status to Interpretation, and confirm that the Interpretation's status changes to 'Provisional' under Dashboard's My Interpretation
  * Note: no Recent History item for the action of adding a Provisional status exists yet


*Screenshots:*
Display of Orpha # in parenthesis:
![image](https://cloud.githubusercontent.com/assets/4326866/19665748/69ebe97a-99fb-11e6-833a-f95420036dbc.png)

Changing of status to Provisional (the only status other than 'In Progress'):
![image](https://cloud.githubusercontent.com/assets/4326866/19665769/8c6120ec-99fb-11e6-8ca2-93f79ffb53c4.png)

New look of Interpretation in Dashboard:
![image](https://cloud.githubusercontent.com/assets/4326866/19665778/99f08c98-99fb-11e6-86ab-508e8cba0b2b.png)

Expanded/improved recent history tracking:
![image](https://cloud.githubusercontent.com/assets/4326866/19665176/974cdb48-99f8-11e6-9b56-7d8f1da5d24c.png)
